src/com/ka/spreadsheet/diff/UnifiedDiffSpreadSheetDiffCallback.java

```diff
@@ -0,0 +1,168 @@
+      "Cell-ordering contract violated.  Previous=" + previousCell.getCellPosition()
+      + ", current=" + c1.getCellPosition();
+    previousCell = c1;
+    accumulateAndMaybePrint(new DiffCell(
+      c1.getSheetName(),
+      c1.getRowIndex(),
+      c1.getColumnIndex(),
+      c1.getCellValue(),
+      c2.getCellValue()
+    ));
+  }
+
+  private void accumulateAndMaybePrint(DiffCell diffCell) {
+    if (!isSameSheet(prevDiffCell, diffCell)) {
+      printAndEmptyCellBlock();
+      System.out.println("--- " + file1 + "!" + diffCell.sheetName);
+      System.out.println("+++ " + file2 + "!" + diffCell.sheetName);
+    }
+    if (!isSameRow(prevDiffCell, diffCell)) {
+      printAndEmptyCellBlock();
+    }
+    if (!isSameCellBlock(prevDiffCell, diffCell)) {
+      printAndEmptyCellBlock();
+    }
+    currentCellBlock.add(diffCell);
+    prevDiffCell = diffCell;
+  }
+
+  private boolean isSameSheet(DiffCell c1, DiffCell c2) {
+    assert c1 != null && c2 != null;
+    return c1.sheetName.equals(c2.sheetName);
+  }
+
+  private boolean isSameRow(DiffCell c1, DiffCell c2) {
+    assert c1 != null && c2 != null;
+    return c1.rowIndex == c2.rowIndex;
+  }
+
+  private boolean isSameCellBlock(DiffCell c1, DiffCell c2) {
+    assert c1 != null && c2 != null;
+    return c1.colIndex == (c2.colIndex - 1);
+  }
+
+  // TODO: Make this handle multiple rows, maybe.  What would that output look like?  This might be a bad idea.
+  private void printAndEmptyCellBlock() {
+    assert currentCellBlock != null;
+    if (currentCellBlock.size() > 0) {
+      StringBuilder sheet1Lines = new StringBuilder();
+      StringBuilder sheet2Lines = new StringBuilder();
+      String cellRange = CELL_INTERNAL_TO_USER(currentCellBlock.get(0).rowIndex, currentCellBlock.get(0).colIndex);
+      if (currentCellBlock.size() > 1) {
+          cellRange = cellRange + "," +
+            CELL_INTERNAL_TO_USER(currentCellBlock.get(0).rowIndex, currentCellBlock.get(currentCellBlock.size()-1).colIndex);
+      }
+      System.out.println("@@ -" + cellRange + " +" + cellRange + " @@");
+      int prevCol = -1;
+      for (DiffCell rowCell : currentCellBlock) {
+        assert currentCellBlock.get(0).rowIndex == rowCell.rowIndex : "printAndEmptyCellBlock() only supports one row at a time.";
+        assert prevCol == -1 || prevCol == (rowCell.colIndex -1 ) : "printAndEmptyCellBlock() only supports a contiguous range of cells at a time.";
+        sheet1Lines.append("-");
+        sheet2Lines.append("+");
+        if (rowCell.c1Value != null) {
+          sheet1Lines.append(rowCell.c1Value);
+        }
+        if (rowCell.c2Value != null) {
+          sheet2Lines.append(rowCell.c2Value);
+        }
+        sheet1Lines.append(lineSeparator);
+        sheet2Lines.append(lineSeparator);
+      }
+      System.out.print(sheet1Lines.toString());
+      System.out.print(sheet2Lines.toString());
+      }
```

Comment:
indentation

Reply:
Heh.  Too many years using simple editors instead of IDEs.  Will do.

Reply:
Done.